src/scores/processing/block_bootstrap.py

```diff
@@ -0,0 +1,410 @@
+
+    # Retrieve sizes of the bootstrap dimensions from objects_list
+    sizes = None
+    for obj in objects_list:
+        try:
+            sizes = OrderedDict(
+                {d: (obj.sizes[d], b) for d, b in blocks.items()},
+            )
+            break
+        except KeyError:
+            pass
+    if sizes is None:
+        raise ValueError(
+            "At least one input object must contain all dimensions in blocks.keys()",
+        )
+
+    # Generate random indices for bootstrapping all objects_list
+    nested_indices = _n_nested_blocked_random_indices(sizes, n_iteration, circular)
+
+    # Expand indices for broadcasting for each object separately
+    indices = []
+    input_core_dims = []
+    for obj in objects_list:
+        available_dims = [d for d in dim if d in obj.dims]
+        indices_to_expand = [nested_indices[key] for key in available_dims]
+
+        indices.append(_expand_n_nested_random_indices(indices_to_expand))
+        input_core_dims.append(available_dims)
+
+    # Process objects_list separately to handle non-matching dimensions
+    result = []
+    for obj, ind, core_dims in zip(objects_list, indices, input_core_dims):
+        if isinstance(obj, xr.Dataset):
+            # Assume all variables have the same dtype
+            output_dtype = obj[list(obj.data_vars)[0]].dtype
+        else:
+            output_dtype = obj.dtype
+
+        result.append(
+            xr.apply_ufunc(
+                _bootstrap,
+                obj,
+                kwargs={"indices": [ind]},
+                input_core_dims=[core_dims],
+                output_core_dims=[core_dims + ["iteration"]],
+                dask="parallelized",
+                dask_gufunc_kwargs={"output_sizes": {"iteration": n_iteration}},
+                output_dtypes=[output_dtype],
+            )
+        )
+
+    # Rename excluded dimensions
+    return tuple(res.rename(rename) for res, rename in zip(result, renames))
+
+
+def block_bootstrap(
+    *objects: XarrayLike,
+    blocks: Dict[str, int],
+    n_iteration: int,
+    exclude_dims: Union[List[List[str]], None] = None,
+    circular: bool = True,
+) -> Union[XarrayLike, Tuple[XarrayLike, ...]]:
+    """
+    Perform block bootstrapping on provided arrays. The function creates new arrays by repeatedly
+    bootstrapping along specified dimensions and stacking the new arrays alone a new "iteration"
+    dimension. Additionally, it includes internal functions for chunk size calculation and
+    handling Dask arrays for chunk size limitation.
+
+    Args:
+        objects: The data to bootstrap, which can be multiple datasets. In the case where
+            multiple datasets are passed, each dataset can have its own set of dimension. However,
+            for successful bootstrapping, dimensions across all input objects must be nested.
+            For instance, for `block.keys=['d1', 'd2', 'd3'], an object with dimension 'd1' and
```

Comment:
```suggestion
            For instance, for ``block.keys=['d1', 'd2', 'd3']``, an object with dimension 'd1' and
```

Note: sphinx inline-code monospacing requires two backticks. There may be other places this may need to be done as well.

Reply:
Done

Reply:
Note: this comment may no longer apply, I vaguely recall seeing someone suggested a config option to allow for '``' to be recognized as monospace/code